mrp_multi_level/models/product_mrp_area.py

```diff
@@ -0,0 +1,302 @@
+    mrp_nbr_days = fields.Integer(
+        string="Nbr. Days",
+        default=0,
+        help="Number of days to group demand for this product during the "
+        "MRP run, in order to determine the quantity to order.",
+    )
+    mrp_qty_multiple = fields.Float(string="Qty Multiple", default=1.00)
+    mrp_transit_delay = fields.Integer(string="Transit Delay", default=0)
+    mrp_verified = fields.Boolean(
+        string="Verified for MRP",
+        help="Identifies that this product has been verified "
+        "to be valid for the MRP.",
+    )
+    mrp_lead_time = fields.Float(string="Lead Time", compute="_compute_mrp_lead_time")
+    distribution_lead_time = fields.Float()
+    main_supplier_id = fields.Many2one(
+        comodel_name="res.partner",
+        string="Main Supplier",
+        compute="_compute_main_supplier",
+        store=True,
+        index=True,
+    )
+    main_supplierinfo_id = fields.Many2one(
+        comodel_name="product.supplierinfo",
+        string="Supplier Info",
+        compute="_compute_main_supplier",
+        store=True,
+    )
+    supply_method = fields.Selection(
+        selection=[
+            ("buy", "Buy"),
+            ("none", "Undefined"),
+            ("manufacture", "Produce"),
+            ("phantom", "Kit"),
+            ("pull", "Pull From"),
+            ("push", "Push To"),
+            ("pull_push", "Pull & Push"),
+        ],
+        compute="_compute_supply_method",
+    )
+    qty_available = fields.Float(
+        string="Quantity Available", compute="_compute_qty_available"
+    )
+    mrp_move_ids = fields.One2many(
+        comodel_name="mrp.move", inverse_name="product_mrp_area_id", readonly=True
+    )
+    planned_order_ids = fields.One2many(
+        comodel_name="mrp.planned.order",
+        inverse_name="product_mrp_area_id",
+        readonly=True,
+    )
+    mrp_planner_id = fields.Many2one("res.users")
+
+    _sql_constraints = [
+        (
+            "product_mrp_area_uniq",
+            "unique(product_id, mrp_area_id)",
+            "The product/MRP Area parameters combination must be unique.",
+        )
+    ]
+
+    @api.constrains(
+        "mrp_minimum_order_qty",
+        "mrp_maximum_order_qty",
+        "mrp_qty_multiple",
+        "mrp_minimum_stock",
+        "mrp_nbr_days",
+    )
+    def _check_negatives(self):
+        values = self.read(
+            [
+                "mrp_minimum_order_qty",
+                "mrp_maximum_order_qty",
+                "mrp_qty_multiple",
+                "mrp_minimum_stock",
+                "mrp_nbr_days",
+            ]
+        )
+        for rec in values:
+            if any(v < 0 for v in rec.values()):
+                raise ValidationError(_("You cannot use a negative number."))
+
+    def name_get(self):
+        return [
+            (
+                area.id,
+                f"[{area.mrp_area_id.name}] {area.product_id.display_name}",
+            )
+            for area in self
+        ]
```

Comment:
@JasminSForgeFlow I see that the name displayed for product mrp areas is broken?

![image](https://github.com/OCA/manufacture/assets/23449160/dd4633e5-3c4c-459b-9327-398a8a9ad16d)

could you check?

Reply:
its fixed,

Thanks